runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -158,11 +160,18 @@ public void write(IToken token, Object data) throws OverwriteException {
         }, true);
 
         // Cache the successful write
-        if (!runtime.getParameters().isCacheDisabled()) {
+        if (!runtime.getParameters().isCacheDisabled() && cacheOption == CacheOption.WRITE_THROUGH) {
             readCache.put(token.getTokenValue(), ld);
         }
     }
 
+    /**
+     * @see AddressSpaceView#write(IToken, Object, CacheOption)
```

Comment:
Add in the comment that this uses WRITE_THROUGH

Reply:
done.